src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -62,17 +74,34 @@ public class HTTPRecordWriter extends RecordWriter<StructuredRecord, StructuredR
   private final HTTPSinkConfig config;
   private StringBuilder messages = new StringBuilder();
   private String contentType;
+  private String url;
+  private String configURL;
+  private List<PlaceholderBean> placeHolderList;
 
   private AccessToken accessToken;
+  private final HttpErrorHandler httpErrorHandler;
+  private final PollInterval pollInterval;
+  private int httpStatusCode;
 
   HTTPRecordWriter(HTTPSinkConfig config) {
     this.config = config;
     this.accessToken = null;
+    this.httpErrorHandler = new HttpErrorHandler(config);
+    if (config.getRetryPolicy().equals(RetryPolicy.LINEAR)) {
+      pollInterval = FixedPollInterval.fixed(config.getLinearRetryInterval(), TimeUnit.SECONDS);
+    } else {
+      pollInterval = IterativePollInterval.iterative(duration -> duration.multiply(2),
+                                                     Duration.FIVE_HUNDRED_MILLISECONDS);
+    }
+    url = config.getUrl();
+    placeHolderList = getPlaceholderListFromURL();
   }
 
   @Override
   public void write(StructuredRecord input, StructuredRecord unused) throws IOException {
     String message = null;
+    messages.setLength(0);
```

Comment:
Setting is 0 for ever write, will clamp message size 1.
This is not compatible with batch logic hence when batchSize is more than 1 this will not write to output any records other than last few.

Reply:
For the subsequent write, the previous messages are getting appended. So, need to reset the variable before every write.

Reply:
write is called for every record.
Say batch size is 2.

- write called for record 1
- messages is set to 0
- record 1 is appended
- check is made if message size == batch size (no as message size is 1 and batch size is 2)
- next write is called for record 2
-  messages is set to 0  

>  record 1 is now lost 😱
> message size will never reach batch size as it's getting reset to 1 for each call

Reply:
As discussed, resetting the messages for only successful requests. 6097e41b01711d5aee749257cc47a270fabee69c